irods/test/data_obj_test.py

```diff
@@ -2024,6 +2024,19 @@ def test_settings_load_and_save_471(self):
                     config.load(**load_logging_options)
                     self.assertTrue(config.data_objects.auto_close, RANDOM_VALUE - i - 1)
 
+    def test_setting_xml_parser_choice_by_environment_only__issue_584(self):
+        program = os.path.join(test_modules.__path__[0], 'test_xml_parser.py')
```

Comment:
Can the assumption about where **we** are ever fail?

Reply:
You're right to be suspicious I think.  Under Python3, `__path__` attribute of a module seems to always contains absolute paths. But Python2 sometimes uses relative paths, which leaves us vulnerable to a chdir call.

Reply:
Further, the only chdir called within the test suite is in sslsetup module, and it's balanced by another chdir in a finally: block which sets the CWD back to what it was before the test.  So, a concern in Python2 only, and then, not so worrying.

Reply:
Can we surround it with an `os.path.abspath` to be sure, if that is indeed a concern?

Reply:
> Can we surround it with an `os.path.abspath` to be sure, if that is indeed a concern?

That might be wise to guard against future attempts to use it in a multithread scenario, yes. Will do.

Reply:
See https://github.com/irods/python-irodsclient/pull/585#issuecomment-2265711900:

> Responding to [#585 (comment)](https://github.com/irods/python-irodsclient/pull/585#discussion_r1684565583) @alanking 's comment. above (Sorry I cannot respond int eh conversation above, but I guess GitHub works in strange ways.)
> 
> The `os.path.absdir` may not be needed, as :
> 
>  1. it would only be needed for Python2, for which we soon plan to drop support
> 
>  2. the only eventuality that would cause the` join`'s result to be inaccurate in finding the target script is an invocation of `os.chdir` by another test, and this will not happen, since setupssl.py is the only thing calling `chdir`, and its only caller (login_auth_test.py) has now [been banished from the test suite](https://github.com/irods/python-irodsclient/commit/459e6434fb49119aa4e59af37815873d435ec742).

Seems reasonable to me to leave as is, but I defer to @korydraughn.

Reply:
Yeah, it's fine.

If something goes wrong, the test will tell us and we'll fix it.

Marking as resolved.